eng/scripts/pre-commit

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+set -e
+
+# Cred Scanner
+
+# Setting up Cred Scanner Pre-Commit Git Hook locally requires installing Guardian and running the
+# initialization first before the pre-commit hook can actually be used.
+# https://eng.ms/docs/products/credential-risk-exposure-defense/solutions/credentials_in_code/precommit-git-hook
```

Comment:
If CredScan doesn't run should we echo this link to explain how to enable CredScan during the pre-commit hook